lib/phubme/comment_parser.ex

```diff
@@ -1,32 +1,15 @@
 defmodule PhubMe.CommentParser do
   def process_comment(body_params) do
     comment = get_in(body_params, ["comment", "body"])
-    sender  = get_in(body_params, ["comment", "user", "login"])
-    comment_source = get_in(body_params, ["comment", "html_url"])
-    IO.puts "Processing comment : \"" <> comment <> "\" from " <> sender
-    comment = comment
-              |> nicknames_present
-              |> extract_nicknames
-    case comment do
-      {:ok, full_comment, nicknames} -> {full_comment, nicknames, sender, comment_source}
-      {:error, _} -> "stop here"
-    end
+    sender = get_in(body_params, ["comment", "user", "login"])
+    source = get_in(body_params, ["comment", "html_url"])
+    nicknames = comment |> extract_nicknames
+    IO.puts "Processing comment : \"#{comment}\" from #{sender}"
+    {comment, nicknames, sender, source}
   end
 
-  defp nicknames_present(comment) do
-    if Regex.match?(~r{@([A-Za-z0-9_]+)}, comment) do
-      {:nicknames_found, comment}
-    else
-      {:no_nicknames_found, comment}
-    end
-  end
-
-  defp extract_nicknames({:nicknames_found, comment}) do
-    nicknames = Regex.scan ~r{@([A-Za-z0-9_]+)}, comment, capture: :first
-    {:ok, comment, nicknames}
-  end
-
-  defp extract_nicknames({:no_nicknames_found, _}) do
-    {:error, "No nicknames found in message"}
+  defp extract_nicknames(comment) do
+    matches = Regex.scan(~r{@([A-Za-z0-9_]+)}, comment, capture: :first)
+    matches |> List.flatten
```

Comment:
Flattening the result of the `scan`, makes things clearer. Instead of `[[nickname], [nickname], ..]`, we ends up with the more intuitive: `[nickname, nickname, ...]` format.